contrib/natss/pkg/dispatcher/dispatcher/dispatcher.go

```diff
@@ -71,11 +76,15 @@ func NewDispatcher(natssURL, clusterID string, logger *zap.Logger) (*Subscriptio
 		clusterID:     clusterID,
 		subscriptions: make(map[provisioners.ChannelReference]map[subscriptionReference]*stan.Subscription),
 	}
-	receiver, err := provisioners.NewMessageReceiver(createReceiverFunction(d, logger.Sugar()), logger.Sugar())
+	receiver, err := provisioners.NewMessageReceiver(
+		createReceiverFunction(d, logger.Sugar()),
+		logger.Sugar(),
+		provisioners.ResolveChannelFromHostHeader(provisioners.ResolveChannelFromHostFunc(d.getChannelReferenceFromHost)))
 	if err != nil {
 		return nil, err
 	}
 	d.receiver = receiver
+	d.setHostToChannelMap(map[string]provisioners.ChannelReference{})
```

Comment:
Should this be before the MessageReceiver is created?

Reply:
why do you think it should be? This is the "New" function.

Reply:
I don't think there will be any ill effects, because as you mentioned the receiver isn't started before this line. It just feels odd to me to set this after the dependent receiver can 'get' it.